pkg/datasource/accessors.go

```diff
@@ -382,6 +384,14 @@ func (a *fieldAccessor) Int64(data Data) int64 {
 	return int64(a.ds.byteOrder.Uint64(val))
 }
 
+func (a *fieldAccessor) Bool(data Data) bool {
+	val := a.Get(data)
+	if len(val) < 1 {
```

Comment:
Now that I think about it, when could this happen?

Reply:
Depends on the origin of the field; if it's a member of a eBPF struct, it usually won't happen as there are other checks in place. If you add a new field manually and didn't reserve space for it, it will be of length 0.

There's room for improvement on how we could internally handle that (e.g. let the DataSource handle memory allocation in the latter case), but that's outside of the scope of this PR :)